File: 2020/62.py
"""
62
"""
import math
class Solution:
    def uniquePaths(self, m: int, n: int) -> int:
        # math
        # the combination of (n-1) "right" and (m-1) "down"
        a = math.factorial(m-1)
        b = math.factorial(n-1)
        return int(math.factorial(m+n-2)/ (a*b))

File: current_session/c++/62.cpp
#include <iostream>

using namespace std;

class Solution {
public:
  int uniquePaths(int m, int n) {
    long ans = 1;
    for(int i = m+n-2, j = 1; i >= max(m, n); i--, j++) 
        ans = (ans * i) / j;
    return ans;
  }
};

int main() {
  cout << Solution().uniquePaths(3,7) << endl;
  cout << Solution().uniquePaths(3,2) << endl;
}
